lib/EnzymeCore/src/EnzymeCore.jl

```diff
@@ -797,4 +798,68 @@ end
 
 Combined(mode::ReverseMode) = mode
 
+"""
+    ChunkStrategy
+
+Abstract type gathering strategies for chunk size selection.
+
+# Subtypes
+
+- [`LargestChunk`](@ref)
+- [`FixedChunk`](@ref)
+- [`AutoChunk`](@ref)
+"""
+abstract type ChunkStrategy end
+
+"""
+    LargestChunk()
+
+Select chunk size equal to the number of elements, so that the corresponding array is processed in a single chunk.
+
+!!! tip
+    In the current Enzyme interface, this strategy is equivalent to setting `chunk = nothing`.
+"""
+struct LargestChunk <: ChunkStrategy end
+
+"""
+    FixedChunk{C}()
+
+Select chunk size equal to a fixed integer `C`.
+
+!!! tip
+    In the current Enzyme interface, this chunk strategy is equivalent to setting `chunk = Val(C)`.
+"""
+struct FixedChunk{C} <: ChunkStrategy end
```

Comment:
for ease of use can we add a constructor FixedChunk(c::Int) = FixedChunk{c}()

Reply:
Is it really a good idea? The constructor `FixedChunk{C}()` explicitly reminds people that `C` is in the type domain, and so that it might be type-unstable to construct it with an arbitrary runtime value. If we define `FixedChunk(C) = FixedChunk{C}()`, we hide that a little bit, that's why I didn't go for it. Ultimately it's your call though

Reply:
Val(x) already has this, I think it's fine here